src/main/java/seedu/address/commons/util/JsonUtil.java

```diff
@@ -68,9 +68,12 @@ public static <T> Optional<T> readJsonFile(
 
         try {
             jsonFile = deserializeObjectFromJsonFile(filePath, classOfObjectToDeserialize);
+        } catch (JsonProcessingException e) {
+            logger.warning("Error processing data from jsonFile file " + filePath + ": " + e);
+            throw new DataConversionException(e);
         } catch (IOException e) {
             logger.warning("Error reading from jsonFile file " + filePath + ": " + e);
-            throw new DataConversionException(e);
+            throw e;
```

Comment:
I've noticed a potential issue with the error handling strategy in this code. Specifically, the practice of catching and immediately re-throwing the same error could lead to some inefficiencies.

In your current setup, you've employed a logger to document exceptions. However, by re-throwing the same exception, it's possible that another component, such as MainApp, might process this exception once more. This could inadvertently lead to duplicate entries in our logs, complicating the debugging process and adding unnecessary clutter to our log files.

To enhance this, I'd suggest revisiting the error handling strategy. If an error is anticipated and can be recovered from, it would be more efficient to handle it directly where it occurs, thus avoiding the need to re-throw it. On the other hand, if the error is unexpected, it might be more beneficial to let it bubble up to a higher level where it can be logged and handled in a more global context. This approach would help us maintain cleaner, more manageable logs and prevent unnecessary operations.

Reply:
Good observations @SPWwj 
I wouldn't worry about the efficiency so much though. The difference in performance is unlikely to matter.
There is value is logging an exception along multiple points between where it was thrown and where it was handled. For one thing, a similar exception can be thrown from multiple places but handled in one place, which means it is useful to record where it was thrown. For another, the decision to log or not is best made locally (i.e., without considering where the method is being used) -- as the usage of a method should be able to change freely without having to change the method. So, we shouldn't make any assumptions about how the method will be used and how the exception will be handled over there.